include/enotify.php

```diff
@@ -452,17 +464,19 @@ function notification($params)
 
 	if ($show_in_notification_page) {
 		$notification = DI::notify()->insert([
-			'name'       => $params['source_name'] ?? '',
-			'name_cache' => substr(strip_tags(BBCode::convert($params['source_name'] ?? '')), 0, 255),
-			'url'        => $params['source_link'] ?? '',
-			'photo'      => $params['source_photo'] ?? '',
-			'link'       => $itemlink ?? '',
-			'uid'        => $params['uid'] ?? 0,
-			'iid'        => $item_id ?? 0,
-			'parent'     => $parent_id ?? 0,
-			'type'       => $params['type'] ?? '',
-			'verb'       => $params['verb'] ?? '',
-			'otype'      => $params['otype'] ?? '',
+			'name'          => $params['source_name'] ?? '',
+			'name_cache'    => substr(strip_tags(BBCode::convert($params['source_name'] ?? '')), 0, 255),
+			'url'           => $params['source_link'] ?? '',
+			'photo'         => $params['source_photo'] ?? '',
+			'link'          => $itemlink ?? '',
+			'uid'           => $params['uid'] ?? 0,
+			'iid'           => $item_id ?? 0,
+			'uri-id'        => $uri_id ?? 0,
+			'parent'        => $parent_id ?? 0,
+			'parent-uri-id' => $parent_uri_id ?? 0,			
```

Comment:
If `$uri_id` and `$parent_uri_id` aren't used anywhere else, you can write this instead and remove the added lines 110-115 and 122-127:
```suggestion
			'uri-id'        => $params['item']['uri-id'] ?? 0,
			'parent'        => $parent_id ?? 0,
			'parent-uri-id' => $params['item']['parent-uri-id'] ?? 0,
```
If they are used before this, you can drop the `??` like this since you ensured these variables were initialized:
```suggestion
			'uri-id'        => $uri_id,
			'parent'        => $parent_id ?? 0,
			'parent-uri-id' => $parent_uri_id,
```
In either case, there seems to be superfluous trailing white spaces on line 476.

Reply:
Blame VSCode for the white spaces.

Reply:
It's okay, you still have to choose either suggestion.

Reply:
I would like to keep the variables to keep it identical to the current coding (although this function really needs an overhaul)

Reply:
The current code contains both intermediary variables and direct references to `$params` in the assignment array so I'm not sure what you mean by "identical".

Reply:
identical with the fields like `$item_id` which serves the same purpose as `$uri_id`.

Reply:
Ok, then the second suggestion it is.

Reply:
Yeah, I already pushed something similar.